### PR TITLE
Build arm64 Python packages

### PR DESCRIPTION
**Pull request by freddan80**

### CODE REVIEW

File: .github/workflows/build_package.yml
Comment (freddan80) [inline .github/workflows/build_package.yml]: Not sure what the managed releaser is/are. Are there options?

File: .github/workflows/build_package.yml
Comment (freddan80) [inline .github/workflows/build_package.yml:101]: This is what is used [here](https://github.com/nod-ai/SRT/blob/b9dd7ee0404071415e8984bfe11406ec51ddb431/.github/workflows/build_package.yml#L85C9-L85C9)

**Review by freddan80 — COMMENTED** [review]
Leaving a few open question marks.

File: .github/workflows/build_package.yml
Comment (ScottTodd) [inline .github/workflows/build_package.yml]: Is an arm64 build necessary for the tf-compiler-tools? Those are pure Python, which might be portable across CPU architecture? You could try installing these from pip to see.

File: .github/workflows/build_package.yml
Comment (ScottTodd) [inline .github/workflows/build_package.yml]: As these steps are identical, you should be able to omit the extra steps and instead check if the "build family" includes "linux" for _either_ x86 or arm. You may not even need to differentiate between `build-family: linux-aarch64` and `build-family: linux-x86_64` at all.

`build_linux_packages.sh` already chooses which Docker image to use, so no need to plumb that through: https://github.com/openxla/iree/commit/e791af1b82c59d278cb4837595c96fa67e477fcd

So to summarize, I think you only need to add
* `Main distribution (Linux-AArch64)` (if people are asking for it, we don't have macOS or Windows "main distributions", fwiw)
* `linux-aarch64` steps under the `matrix: include:` at the top of the file that use aarch64 runners to build the linux steps

File: .github/workflows/build_package.yml
Comment (ScottTodd) [reply .github/workflows/build_package.yml:101]: and also here: https://github.com/openxla/iree/blob/988f7c560f484640e910af6f284fa08565bf98dc/build_tools/python_deploy/build_linux_packages.sh#L67

File: .github/workflows/build_package.yml
Comment (ScottTodd) [reply .github/workflows/build_package.yml]: @stellaraccident do you know how the "managed-releaser" runners are different?

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml]: Good question. I thought about it and decided to add it since I wasn't sure it was guaranteed to be pure Python going fwd. "Might as well...". I'll try though.

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml:101]: Added a comment about that. I can lift it out as a parameter you prefer (to have these dockers in one place)? LMKWYT

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml]: Fixed. I should have seen that :relieved: 

> if people are asking for it, we don't have macOS or Windows "main distributions", fwiw

I kept it for now. Is there a drawback to keep it?

Comment (ScottTodd) [reply .github/workflows/build_package.yml]: Possible drawbacks:

* Workflow file complexity
* Extra build time
* Potential for build instability (real build failures or flakes)

File: .github/workflows/build_package.yml
Comment (ScottTodd) [inline .github/workflows/build_package.yml]: How about shortening these checks to this?

* It's a bit over 80 characters, but we don't use line length limits in .yml files
* Less risk of a typo, and naturally extends to any future `linux-*` builds

```suggestion
        if: "matrix.build-package == 'py-runtime-pkg' && contains(matrix.build-family, 'linux')"
```

Then you could either match the 'contains' style for other platforms or just revert the other diffs

```diff
-        if: "matrix.build-package == 'py-runtime-pkg' && matrix.build-family == 'windows'"
+        if: "matrix.build-package == 'py-runtime-pkg' && contains(matrix.build-family, 'windows')"
```

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml]: Ok, seems to me like the drawbacks aren't that burdensome, but it might be me overlooking something. OTOH, if nobody is asking for it there's no point in doing it.... Unfortunately I don't have idea whether this is being asked for, so I rely on your judgement there. WDYT?

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml]: Ok, will fix

**Review by ScottTodd — COMMENTED** [review]
Okay, this LGTM once we figure out what to do with 'managed-releaser'

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml]: @stellaraccident or someone else with access to the runners. What are the managed-releaser? Cheers

**Comment by freddan80** [comment]
@ScottTodd @stellaraccident, is there anything I can do to help progress this further? Who has insight to the managed-releasers?

I'd be keen on finishing #12684 asap, an I think this is what's lefty. If it's not too much trouble thx

File: .github/workflows/build_package.yml
Comment (stellaraccident) [inline .github/workflows/build_package.yml:50]: I believe they are merely "different" - as in a dedicated instance that is not shared with the presubmit pool. That is done for security as opposed to a functional requirement.

I'm interested in seeing this landed but have been having trouble following the deps on different threads. Can we start a thread on discord and coordinate there?

File: .github/workflows/build_package.yml
Comment (stellaraccident) [reply .github/workflows/build_package.yml]: I believe they are merely "different" - as in a dedicated instance that is not shared with the presubmit pool. That is done for security as opposed to a functional requirement.

I'm interested in seeing this landed but have been having trouble following the deps on different threads. Can we start a thread on discord and coordinate there?

**Review by stellaraccident — APPROVED** [review]

File: .github/workflows/build_package.yml
Comment (freddan80) [reply .github/workflows/build_package.yml:50]: :+1:

File: .github/workflows/build_package.yml
Comment (pzread) [inline .github/workflows/build_package.yml]: If we want to move forward with postsubmit runner:

```suggestion
          - runs-on: [self-hosted, arm64, os-family=Linux, runner-group=postsubmit]
```

Comment (freddan80) [reply .github/workflows/build_package.yml]: I forgot why we'd run the postsubmit runner..? IIUC postsubmit won't run on PR's. Is it for initial stability?

Comment (freddan80) [reply .github/workflows/build_package.yml]: I talked to @ScottTodd and we agreed it would be a good idea to set `experimental: true` such that a failing build don't stop fail the job. Also for stability. Let me know what you think.

Comment (stellaraccident) [reply .github/workflows/build_package.yml]: Yes, please. Will be easier to iterate on that way.

Comment (stellaraccident) [reply .github/workflows/build_package.yml]: This workflow only runs on scheduled release jobs or manual triggering. In both cases, it runs on committed patches (or those from folks with write access, which can be more trusted).

Comment (pzread) [reply .github/workflows/build_package.yml]: > I forgot why we'd run the postsubmit runner..? IIUC postsubmit won't run on PR's. Is it for initial stability?

Since this is to build the release package, postsubmit runners are safer to use as presubmit runners might run and access unverified code.

If you want to test the workflow in PR, you can change it to `presubmit` temporarily (and it will fail when trying to publish the package). Just need to change it back to postsubmit before merge

Comment (freddan80) [reply .github/workflows/build_package.yml]: Pushed `experimental: true` (I removed the commit) and then a patch for using `postsubmit` runner group. Then added a patch `presubmit` to try things out temporarily (will be reverted). I expect it to build fine, since I've tested it locally, but you never know...

**Comment by stellaraccident** [comment]
I think we just need to land and test in prod. With experimental set to true, it won't block anything and then we can iterate. If you don't have an objection, I'll just do that and land it now. Then we can see if it worked in the morning.

**Comment by stellaraccident** [comment]
Ok, landed with the revert to the prior commit. We can check on it in the morning and see that everything worked. There should still be time tonight for the CI to turn green before the release is cut (the release will only cut from a green patch on main).

**Comment by freddan80** [comment]
Nice! I'll check how it went later today. Thx for all the help on this, I know it's late there now :+1:

**Pull request merged**
